app/setup/index.js

```diff
@@ -0,0 +1,259 @@
+                <h3 className="block-center-heading">
+                    Choose Your Plotly Use Case
+                </h3>
+                    <div className="controls" style={{padding: '20px'}}>
+                        {submitButton(CLOUD)}
+                        {submitButton(ONPREM)}
+                    </div>
+                    <span>
+                        {'Learn more about our services '}
+                        <a href="https://plot.ly/products/cloud/">
+                        {'here'}</a>.
+                    </span>
+                </div>
+            );
+        };
+
+        const loginCloud = () => {
+            return (
+                <div className="control-group">
+                    <h3 className="block-center-heading">{'Login Into Your Account'}</h3>
+                    <div className="controls" style={{padding: '20px'}}>
+                        <div className="form-group">
+                            <label>Your Plotly Username</label>
+                            <input
+                                type="text"
+                                className="form-control"
+                                placeholder="johndoe"
+                                name="username"
+                                onChange={(e) => this.updateStateWithEvent(e)}
+                            ></input>
+                            {loginButton()}
+                        </div>
+                    </div>
+                </div>
+
+            );
+        };
+
+        const loginOnPrem = () => {
+            return (
+                <div className="control-group">
+                    <h3 className="block-center-heading">{'Login Into Your Account'}</h3>
+                    <div className="controls" style={{padding: '20px'}}>
+                        <div className="form-group">
+                            <label>Your On-Prem Plotly Username</label>
+                            <input
+                                type="text"
+                                className="form-control"
+                                placeholder="johndoe"
+                                name="username"
+                                onChange={(e) => this.updateStateWithEvent(e)}
+                            ></input>
+                            <label>Your On-Prem Plotly Domain</label>
+                            <input
+                                type="text"
+                                className="form-control"
+                                placeholder="plotly.your-company.com"
+                                name="domain"
+                                onChange={(e) => this.updateStateWithEvent(e)}
+                            ></input>
+                            {loginButton()}
+                        </div>
+                    </div>
+                </div>
+            );
+        };
+
+        const loginOptions = {
+            cloud: loginCloud(),
+            onprem: loginOnPrem()
```

Comment:
if you're just going to instantiate both of these, why even make then functions in the first place?